### PR TITLE
Update docs with details about reusable workflow.

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

During the work on #9976 we have learned that OIDC token doesn't contain all required data to clone proper source code version in oci-image-builder pipeline. This is a blocker for image-builder github workflow integration. This proposal describe how we can use a reusable workflow to collect all required data and pass them in secure way to the oci-image-builder pipeline.

### CODE REVIEW

File: cmd/image-builder/github-workflow-integration.md [hunk 1]
```diff
@@ -36,6 +46,24 @@ process is executed in an Azure DevOps pipeline, providing an SLC-29-compliant i
...
+
+Reusable workflow image-builder is a GitHub workflow
+that is used to collect required data from workflow inputs and GitHub context variables,
+retrieve the OIDC token from GitHub's OIDC identity provider, and trigger the `oci-image-builder` pipeline in Azure DevOps.
+Because the oidc token does not contain all the required data for the `oci-image-builder` pipeline,
+the reusable workflow image-builder collects additional data from GitHub context variables.
+The oidc token alone does not contain enough data to clone the appropriate source code for the build process.
+Using the reusable workflow
+we bundle all the steps required to collect the data and trigger the `oci-image-builder` pipeline in a controlled and secure enviornment.
+Using an OIDC token allows us to confirm the version and identity of the workflow that triggered the build process.
```
Comment (IwonaLanger) [inline cmd/image-builder/github-workflow-integration.md:59]:
```suggestion
Using an OIDC token allows you to confirm the version and identity of the workflow that triggered the build process.
```

File: cmd/image-builder/github-workflow-integration.md [hunk 2]
```diff
@@ -36,6 +46,24 @@ process is executed in an Azure DevOps pipeline, providing an SLC-29-compliant i
...
+The reusable workflow is stored in the `github.com/kyma-project/test-infra` repository in the main branch and changes to the workflow are
+versioned
+and provided using pull requests.
```
Comment (IwonaLanger) [inline cmd/image-builder/github-workflow-integration.md:62]:
```suggestion
The reusable workflow is stored in the `github.com/kyma-project/test-infra` repository on the `main` branch, and changes to the workflow are
versioned
and provided using pull requests.
```

File: cmd/image-builder/github-workflow-integration.md [hunk 3]
```diff
@@ -105,14 +131,11 @@ These parameters include user-defined parameters, parameters from the OIDC token
...
 - **RepoOwner**: The owner of the repository. Possible values include `kyma-project` and `kyma-incubator`.
 - **JobType**: The type of job. Possible values include `presubmit` and `postsubmit`.
-- **PullBaseSHA**: The base SHA of the pull request.
-- **PullPullSHA**: The SHA of the pull request.
-
-### Computed Parameters
 
-The data for this parameter is not available in the OIDC token and should not be provided by the user.
-We can extract it from the `ref` claim of the OIDC token.
+### Parameters GitHub Context Variables
```
Comment (IwonaLanger) [inline cmd/image-builder/github-workflow-integration.md:135]: Is this heading correct? Isn't anything missing?

File: cmd/image-builder/github-workflow-integration.md [hunk 4]
```diff
@@ -36,6 +46,24 @@ process is executed in an Azure DevOps pipeline, providing an SLC-29-compliant i
...
+## Reusable Workflow Image Builder
+
+Reusable workflow image-builder is a GitHub workflow
+that is used to collect required data from workflow inputs and GitHub context variables,
+retrieve the OIDC token from GitHub's OIDC identity provider, and trigger the `oci-image-builder` pipeline in Azure DevOps.
+Because the oidc token does not contain all the required data for the `oci-image-builder` pipeline,
+the reusable workflow image-builder collects additional data from GitHub context variables.
+The oidc token alone does not contain enough data to clone the appropriate source code for the build process.
+Using the reusable workflow
+we bundle all the steps required to collect the data and trigger the `oci-image-builder` pipeline in a controlled and secure enviornment.
```
[the remark this answers is not in the document]
Comment (dekiel) [reply cmd/image-builder/github-workflow-integration.md:58]: This doc and this part is more about how we as a team are going to solve the problem and design a solution, not some user will or have to do. Do you think form you still is valid in such case?

[the remark this answers is not in the document]
Comment (IwonaLanger) [reply cmd/image-builder/github-workflow-integration.md:58]: No, it's not valid then. Let's keep `we`.

File: cmd/image-builder/github-workflow-integration.md [hunk 5]
```diff
@@ -7,22 +7,32 @@ process is executed in an Azure DevOps pipeline, providing an SLC-29-compliant i
...
-   and the image to build.
-
-4. **Validating the OIDC Token**: The `oci-image-builder` pipeline, running in Azure DevOps, detects a call originating from GitHub and
-   validates the OIDC token against GitHub's
-   OIDC identity provider. This step ensures that the token is valid and has not been tampered with.
-
-5. **OCI Image build preparation**: The `oci-image-builder` pipeline uses the information from the OIDC token to clone the appropriate
+1. **Trigger workflow**: The user or automation triggers a GitHub workflow. The workflow calls reusable workflow image-builder to build the
+   OCI image. The image-builder reusable workflow is stored in the github.com/kyma-project/test-infra repository in the main branch. The
+   reusable workflow defines inputs which are used to pass user-defined parameters to the `oci-image-builder` pipeline.
```
[the remark this answers is not in the document]
Comment (IwonaLanger) [reply cmd/image-builder/github-workflow-integration.md:12]: Sure, how about 'the image-builder reusable workflow'?

File: cmd/image-builder/github-workflow-integration.md [hunk 6]
```diff
@@ -36,6 +46,24 @@ process is executed in an Azure DevOps pipeline, providing an SLC-29-compliant i
    service to sign the OCI image.
    This step ensures the integrity and authenticity of the OCI image.
 
+## Reusable Workflow Image Builder
```
[the remark this answers is not in the document]
Comment (IwonaLanger) [reply cmd/image-builder/github-workflow-integration.md:49]: How about 'the image-builder reusable workflow'?

File: cmd/image-builder/github-workflow-integration.md (hunk 1 above)
Comment (IwonaLanger) [reply cmd/image-builder/github-workflow-integration.md:59]: Same here, let's keep `us`.

File: cmd/image-builder/github-workflow-integration.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (dekiel) [reply cmd/image-builder/github-workflow-integration.md:49]: Will align docs.

File: cmd/image-builder/github-workflow-integration.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (dekiel) [reply cmd/image-builder/github-workflow-integration.md]: Aligned

File: cmd/image-builder/github-workflow-integration.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (dekiel) [reply cmd/image-builder/github-workflow-integration.md:48]: done

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**